resource/opentelemetry-resource-detector-container/src/opentelemetry/resource/detector/container/__init__.py

```diff
@@ -0,0 +1,89 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+from opentelemetry.semconv.resource import ResourceAttributes
+
+logger = logging.getLogger(__name__)
+DEFAULT_CGROUP_V1_PATH = "/proc/self/cgroup"
+DEFAULT_CGROUP_V2_PATH = "/proc/self/mountinfo"
+CONTAINER_ID_LENGTH = 64
+
+
+def _get_container_id_v1():
+    container_id = None
+    try:
+        with open(
+            DEFAULT_CGROUP_V1_PATH, encoding="utf8"
+        ) as container_info_file:
+            for raw_line in container_info_file.readlines():
+                line = raw_line.strip()
+                if len(line) > CONTAINER_ID_LENGTH:
+                    container_id = line[-CONTAINER_ID_LENGTH:]
+                    break
+    except FileNotFoundError as exception:
+        logger.warning("Failed to get container id. Exception: %s", exception)
+    return container_id
+
+
+def _get_container_id_v2():
```

Comment:
Consider using regex, and maybe there is the same pattern for v1 and v2

Reply:
The same logic was used in node.js agent as well.
https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts
I thought of following the same logic here